### ISSUE TITLE
Error: Metadata version mismatch for module

### ISSUE THREAD

**Issue by MSkraburski** [fegyi001/mangol#17]
I tried to follow the manual to install the plugin but end up with that plot of errors. I think I missed something pretty obvious but I do not know what I shall do to make it work properly. 
Could anybody help?

`ERROR in Error: Metadata version mismatch for module xxxxx/node_modules/mangol/node_modules/@angular/core/core.d.ts, found version 4, expec
ted 3, resolving symbol MangolModule in xxxxx/node_modules/mangol/src/lib/modules/_index.ts, resolving symbol MangolModule in xxxxxx/node_modules/mangol/src/lib/modules/_index.ts
    at Error (native)
    at syntaxError (xxxxx\node_modules\@angular\compiler\bundles\compiler.umd.js:1729:34)
    at simplifyInContext (xxxx\node_modules\@angular\compiler\bundles\compiler.umd.js:25111:23)
    at StaticReflector.simplify (xxxx\node_modules\@angular\compiler\bundles\compiler.umd.js:25123:13)
    at StaticReflector.annotations (xxxx\node_modules\@angular\compiler\bundles\compiler.umd.js:24553:41)
    at _getNgModuleMetadata (xxxxx\node_modules\@angular\compiler-cli\src\ngtools_impl.js:138:31)
    at _extractLazyRoutesFromStaticModule (xxxxx\node_modules\@angular\compiler-cli\src\ngtools_impl.js:109:26)
    at xxxxx\node_modules\@angular\compiler-cli\src\ngtools_impl.js:129:27
    at Array.reduce (native)
    at _extractLazyRoutesFromStaticModule (xxxxx\node_modules\@angular\compiler-cli\src\ngtools_impl.js:128:10)
    at includeLazyRouteAndSubRoutes (xxxxx\node_modules\@angular\compiler-cli\src\ngtools_impl.js:66:25)
    at Array.reduce (native)
    at Object.listLazyRoutesOfModule (xxxxx\node_modules\@angular\compiler-cli\src\ngtools_impl.js:54:36)
    at Function.NgTools_InternalApi_NG_2.listLazyRoutes (xxxxx\node_modules\@angular\compiler-cli\src\ngtools_api.js:91:39)
    at AotPlugin._getLazyRoutesFromNgtools (xxxx\node_modules\@ngtools\webpack\src\plugin.js:212:44)
    at _donePromise.Promise.resolve.then.then.then.then.then (xxxxx\node_modules\@ngtools\webpack\src\plugin.js:448:24)
    at process._tickCallback (internal/process/next_tick.js:109:7)
`

**Comment by fegyi001** [comment]
I would try to delete my node_modules dir, then reinstall the dependencies with "npm install". Maybe this solves the issue.

**Comment by pmeems** [comment]
I have the same problem. I don't want to upgrade to Angular5.
How to fix this? I'm really stuck now.

This is my package.json:

    "@angular/animations": "^4.1.0",
    "@angular/cdk": "^5.0.0-rc0",
    "@angular/cli": "^1.5.4",
    "@angular/common": "^4.1.0",
    "@angular/compiler": "^4.1.0",
    "@angular/compiler-cli": "^4.1.0",
    "@angular/core": "^4.1.0",
    "@angular/forms": "^4.1.0",
    "@angular/http": "^4.1.0",
    "@angular/material": "^5.0.0-rc0",
    "@angular/platform-browser": "^4.1.0",
    "@angular/platform-browser-dynamic": "^4.1.0",
    "@angular/platform-server": "^4.1.0",
    "@angular/router": "^4.1.0",
    "@types/node": "^6.0.70",
    "angular-confirmation-popover": "^3.1.1",
    "angular2-color-picker-mani": "^1.6.0",
    "core-js": "^2.4.1",
    "jquery": "^3.2.1",
    "mangol": "latest",
    "ng2-datetime": "^1.4.0",
    "ngx-popover": "0.0.16",
    "rxjs": "^5.3.0",
    "ts-helpers": "^1.1.1",
    "typescript": "^2.3.2",
    "zone.js": "^0.8.4"

**Comment by fegyi001** [comment]
Why do you stick to Angular4?
By the way, I completely refactored the project, please try the latest version (1.0.0 at the moment). Hopefully the error will disappear.

**Comment by fegyi001** [comment]
@pmeems @MSkraburski Do you still experience the aforementioned issue with the refactored version of Mangol? The latest version is now 1.0.3

**Comment by pmeems** [comment]
I've just upgraded my app to Angular5, CLI v1.7 and Mangol v1.0.3
Now I'm getting a different error. This time about MapSkin:

```
ERROR in ./node_modules/raw-loader!./node_modules/postcss-loader/lib??embedded!./node_modules/sass-loader/lib/loader.js??ref--8-3!./src/styles.scss
(Emitted value instead of an instance of Error) CssSyntaxError: J:\MyApp.Web\node_modules\mapskin\css\mapskin.min.css:8:72: Can't resolve '../node_modules/mapskin/fonts/mapskin.eot)%20format(%22embedded-o
pentype%22),url(../node_modules/mapskin/fonts/mapskin.ttf)%20format(%22truetype%22),url(../node_modules/mapskin/fonts/mapskin.woff)%20format(%22woff%22),url(../node_modules/mapskin/fonts/mapskin.svg)%20format(%22svg%22' in 'J:\MyApp.Web\src'
```

**Comment by pmeems** [comment]
Downgrading to Angular CLI v1.6.8 solved the issue about the MapSkin.css

**Comment by fegyi001** [comment]
Thanks for the comment. I just now have fixed this in the latest release (1.0.4), you don't have to downgrade your angular-cli. Mapskin is from now on imported from its official cdn, and this eliminates this issue.

Please try it out if you have time! Thanks!
